src/otx/algo/utils/xai_utils.py

```diff
@@ -101,10 +103,62 @@ def convert_maps_to_dict_image(saliency_maps: np.array) -> list[dict[Any, np.arr
     return [{"map_per_image": map_per_image} for map_per_image in saliency_maps]
 
 
-def postprocess(saliency_map: np.ndarray, output_size: tuple | list = (224, 224)) -> np.ndarray:
+def postprocess(saliency_map: np.ndarray, output_size: tuple | None) -> np.ndarray:
     """Postprocess single saliency map."""
     if saliency_map.ndim != 2:
         raise ValueError
 
-    saliency_map = cv2.resize(saliency_map, output_size)
+    if output_size:
+        h, w = output_size
+        saliency_map = cv2.resize(saliency_map, (w, h))
     return cv2.applyColorMap(saliency_map, cv2.COLORMAP_JET)
+
+
+def dump_saliency_maps(
+    predict_result: list[Any] | list[OTXBatchPredEntityWithXAI | InstanceSegBatchPredEntityWithXAI],
+    explain_config: ExplainConfig,
+    datamodule: EVAL_DATALOADERS | OTXDataModule,
+    output_dir: Path,
+    weight: float = 0.3,
+) -> None:
+    """Sumps saliency maps (raw and with overlay)."""
+    output_dir = output_dir / "saliency_maps"
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    for predict_result_per_batch in predict_result:
+        saliency_maps = predict_result_per_batch.saliency_maps
+        imgs_info = predict_result_per_batch.imgs_info
+        for pred_index in range(len(saliency_maps)):
+            img_id = imgs_info[pred_index].img_idx
+            img_data, image_save_name = _get_image_data_name(datamodule, img_id)
+
+            for class_id, s_map in saliency_maps[pred_index].items():
+                file_name_map = Path(image_save_name + "_class_" + str(class_id) + "_saliency_map.png")
```

Comment:
Actually, can we use here '.jpg' format instead of '.png' since it saves some memory? Just from my experience, after saving a lot of saliency maps I had memory issues

Reply:
Well, it is not that clear that ".jpg" would be better.

Usually, xai model analysis is done manually by a human. Which means that the number of inspecting images is limited (up to hundreds let's say). Which is not that dramatic in terms if memory. If the user accidently runs explain for a million of images, ".jpg" is not gonna save his memory (jpg it still scales linearly).
With that being said, and assuming that ".jpg" is a lossy compression, I would propose to keep png. What do you think?

Reply:
Okay, let's keep it as it is. I actually had the memory problem while compiling presentation with different saliency map results. But it can be called a pretty rare usecase.